Sources/ClientGtk/ServerListView.swift

```diff
@@ -41,61 +46,135 @@ struct ServerListView: View {
     } detail: {
       VStack {
         switch state.detailState {
-          case .addition:
-            additionView
-          case .server(let server):
-            Text(server.name)
-            if let port = server.port {
-              Text("\(server.host):\(port)")
-            } else {
-              Text(server.host)
-            }
-
-            Button("Connect") {
-              completionHandler(server)
-            }
-          case .error(let message):
+          case .server(let index):
+            serverView(index)
+          case .adding:
+            addingView()
+          case .editing(let index):
+            editingView(index)
+          case .error(let message, let returnState):
             Text("Error: \(message)")
-            Button ("Back") {
-              state.detailState = .addition
+            if let returnState {
+              Button ("Back") {
+                state.detailState = returnState
+              }
             }
         }
       }
       .padding(.leading, 10)
     }
   }
 
-  var additionView: some View {
-    VStack {
+  func serverView(_ index: Int) -> some View {
+    let server = state.servers[index]
+
+    return VStack {
+      Text(server.name)
+      if let port = server.port {
+        Text("\(server.host):\(port)")
+      } else {
+        Text(server.host)
+      }
+
+      Button("Connect") {
+        completionHandler(server)
+      }
+
+      Button("Edit") {
+        state.name = server.name
+        if let port = server.port {
+          state.address = "\(server.host):\(port)"
+        } else {
+          state.address = "\(server.host)"
+        }
+        state.detailState = .editing(index)
+      }
+    }
+  }
+
+  func addingView() -> some View {
+    return VStack {
       Text("Add server")
       TextField("Name", state.$name)
       TextField("Address", state.$address)
       
       Button("Add") {
-        let parts = state.address.split(separator: ":")
-        guard parts.count > 0, parts.count <= 2 else {
-          state.detailState = .error("Invalid address")
-          return
+        do {
+          let (host, port) = try ServerListView.parseAddress(state.address)
```

Comment:
To call a static method on the same type as the current method you can use `Self`;

`Self.parseAddress(state.address)`